### DISCUSSION TITLE
QTable - Search in expanding rows text

### DISCUSSION THREAD

**Discussion by aberbenni** [quasarframework/quasar#16006] [General - Components / Directives / etc]
Is there an option to search in [expanding rows](https://quasar.dev/vue-components/table#expanding-rows) text within table?

**Comment by aberbenni** [comment]
A the moment I'm using a hidden column in this tricky way:

```
{
    name: "desc",
    label: "Dessert (100g serving)",
    field: "name",
    required: true,
    style: 'display: none',
}
```

```
<template v-slot:top-right>
   <q-select v-model="visibleColumns" multiple outlined :options="columns.filter((v, i) => v.name != 'desc')" />
</template>
```

```
<q-tr v-show="props.expand" :props="props">
  <q-td colspan="100%">
    <div class="text-left">{{ props.row.name }}</div>
  </q-td>
</q-tr>

```

**Comment by metalsadman** [comment]
Use filter-method

**Reply by aberbenni** [reply]
Yes of course, that is an option.
But I think it shouldn't be necessary to _reinvent the wheel_ just to search in a _default_ part of the table.
Rather it might be useful to handle a "_ghost column_" as an _expanding row_, sort of like I did but in a more standardized way.

**Reply by metalsadman** [reply]
It's there so devs can control the filter behavior since the internal filter function only searches for the properties defined in the column. The filter function can be found here https://github.com/quasarframework/quasar/blob/dev/ui/src/components/table/table-filter.js#L8.
